### PR TITLE
Toast: Add role of status for accessibility

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Kapture 2021-08-30 at 10 27 52
### Summary

Based on feedback from Marcy Sutton's audit. Ensure Toasts announce themselves to a screen reader.

### Links

- [Jira](https://jira.pinadmin.com/browse/PDS-2950)

### Checklist

- [X] Updated unit and Flow Tests
- [X] Verified accessibility: keyboard & screen reader interaction

### CODE REVIEW

File: docs/pages/design_tokens.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ card(
               <Text weight="bold">CSS Token Name</Text>
             </Table.HeaderCell>
             <Table.HeaderCell>
-              <Text weight="bold">JaveScript Prop Name</Text>
+              <Text weight="bold">Javescript Prop Name</Text>
```
Comment (AlbertCarreras) [inline docs/pages/design_tokens.js:29]: Javascript typo

File: packages/gestalt/src/Toast.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ export default function Toast({
   thumbnailShape = 'square',
 }: Props): Node {
   return (
-    <Box marginBottom={3} paddingX={4} maxWidth={360} width="100vw">
+    <Box marginBottom={3} paddingX={4} maxWidth={360} width="100vw" role="status">
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Toast.js:27]: main change!

**Review by AlbertCarreras — APPROVED** [review]
LGTM

File: docs/pages/design_tokens.js (hunk 1 above)
Comment (ayeshakmaz) [reply docs/pages/design_tokens.js:29]: hah, fixed one typo and missed the other - good catch!

File: docs/pages/design_tokens.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ card(
               <Text weight="bold">CSS Token Name</Text>
             </Table.HeaderCell>
             <Table.HeaderCell>
-              <Text weight="bold">JaveScript Prop Name</Text>
+              <Text weight="bold">JavaScript Prop Name</Text>
```
Comment (ayeshakmaz) [inline docs/pages/design_tokens.js:29]: typo fix

**Comment by dangerismycat** [comment]
just an FYI: this would've been better as two or even three PRs, to make sure that each PR is focused on accomplishing just one thing. this PR is fine, but in the future we should try to stick with more narrowly-scoped PRs 🙂

**Pull request merged**